src/hydration/react.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import {
+  useQueryCache,
+  ReactQueryCacheProvider as CacheProvider,
+} from 'react-query'
+import { hydrate } from './hydration'
+
+import type { HydrateConfig } from './hydration'
+import type { ReactQueryCacheProviderProps } from '../react'
+
+export function useHydrate(queries: unknown, hydrateConfig?: HydrateConfig) {
```

Comment:
With the `ReactQueryCacheProvider` component, think we can make this one private?

Reply:
I thought about that, but I think there are still usecases for this one, for example hydrating into an existing cache.

The current Next example is implemented entirely in `_app.js`, but there are also cases where you might want to hydrate in the page components instead (to not rely on a magical prop name for example).

Reply:
Would it then be possible to only have `useHydrate` instead of `ReactQueryCacheProvider`? Then people can either do `hydrate(queryCache, state, config)` or `useHydrate(queryCache, state, config)`? Just trying to limit the number of possibilities if possible

Reply:
Most people are going to want to use `<ReactQueryCacheProvider dehydratedState={...}>`, so they'll end up implementing it themselves anyway. See earlier discussions in this PR and also, check out the Next implementation example in the docs: https://react-query-j5pum9ymn.vercel.app/docs/guides/ssr#integrating-with-nextjs

You can't `useHydrate` in `_app.js` because you don't have a `queryCache` on context yet, which is why `hydration/ReactQueryCacheProvider` was born. :)